State on video

henpe071 opened the pull request:
Gave the video its own state which will update directly when the sliders in the manipulatorwidget are changed.

The apply button will update the current selected tag_frame or all tag_frames in the current label depending on what is selected in the project tree. It will also be disabled when the values haven't changed.

henpe071 committed:
videoproject have its own state

Video project have its own state. This will give the video a state that it will jump back to when clicking the video again. This state is only updated if the video is the last thing clicked in the project tree, jumping to a tag will not update it

TODO save color corrections as well

henpe071 committed:
color corrections are saved in the video state

henpe071 committed:
Zoom state are saved in the video state

henpe071 committed:
Merge branch 'development' into state-on-video

henpe071 committed:
state video

henpe071 committed:
Apply btn updates tagframe or tag with the changes

Cleaned up everything and added some comments

niklasniklas commented:
Bugs:
Didn't get the Apply for a label to work

Other comments (feature changes):
* The video and all of its' analysies should share the same state.
* All tags (both ordinalry and drawingtags) should have the same behaviour.
* If you are on a label or the "Drawing tag" and change brightness and press apply - all tags under it should be updated with the new brightness. Do not update zoom. 
* Every single tag (or drawingtag) should have a "update tag" with similar behaviour when re-tagging a tag using "T". This also updates zoom.

henpe071 committed:
Drawing tag works as a normal tag

niklasniklas commented:
* Retagging av tag ("T") should also update brightness and contrast

* The label should not have its own brightness and contrast. 
** You want to be able to stand on a tag and adjust it (to a good b&c setting)
** then select the label and press apply for it to be applied to all tags
* If there is no "temporary" setting for the label to read, use the parent-video settings

henpe071 committed:
Added update tag option in context menu

Added a new update tag function and changed tree item clicked to item changed instead which have the side effect of making the arrowkeys work in the project tree

henpe071 committed:
Changed name to tree item changed

henpe071 committed:
Added the last fixes

henpe071 commented:
Think I have fixed everything as wanted. However, as a side effect of fixing the update tag in the context menu, you can now use the arrow keys to navigate throught the project tree.
